java-manta-client-unshaded/src/main/java/com/joyent/manta/http/MantaSSLConnectionSocketFactory.java

```diff
@@ -70,10 +77,35 @@ public MantaSSLConnectionSocketFactory(final ConfigContext config) {
     }
 
     /**
+     * @param config configuration context containing SSL config params
      * @return reference to SSL Context
      */
-    private static SSLContext buildContext() {
-        return SSLContexts.createDefault();
+    private static SSLContext buildContext(final ConfigContext config) {
+        if (config.tlsInsecure()) {
+            try {
+                LOG.warn("Configuration: tlsInsecure is true.  ALL TLS VERIFICATION IS DISABLED!");
+                return SSLContextBuilder.create()
+                    .loadTrustMaterial(new TrustAllStrategy())
+                    .build();
+            } catch (KeyManagementException | KeyStoreException | NoSuchAlgorithmException e) {
+                LOG.error("error while disabling TLS security", e);
+                throw new RuntimeException(e);
```

Comment:
Can you please change the exception type such that it is inheriting from `MantaException`?